ampproject/amphtml
# ✨ [bento-mega-menu] Implemented bento mega menu

# What
Implements the `<bento-mega-menu>` component

### Example of the React API:
```jsx
      <BentoMegaMenu {...args}>
        <BentoMegaMenu.Item>
          <BentoMegaMenu.Title>Menu Item 1</BentoMegaMenu.Title>
          <BentoMegaMenu.Content>
            These are the menu contents
          </BentoMegaMenu.Content>
        </BentoMegaMenu.Item>
        <BentoMegaMenu.Item>
          <BentoMegaMenu.Title>Menu Item 2</BentoMegaMenu.Title>
          <BentoMegaMenu.Content>
            Alias aspernatur beatae deserunt error esse eveniet excepturi quia
            suscipit totam vero.
          </BentoMegaMenu.Content>
        </BentoMegaMenu.Item>
        <BentoMegaMenu.Item>
          <BentoMegaMenu.Title>Menu Item 3</BentoMegaMenu.Title>
          <BentoMegaMenu.Content>
            <p>
              Hic impedit iste iure maiores minus necessitatibus nihil non
              numquam obcaecati pariatur perferendis quisquam quos recusandae
              saepe similique tempora ullam velit voluptatibus.
            </p>
          </BentoMegaMenu.Content>
        </BentoMegaMenu.Item>
      </BentoMegaMenu>
```


# Testing 
1. Run `amp storybook`
2. Open the `MegaMenu` item and play with it:

https://user-images.githubusercontent.com/430608/168641118-addadbde-7a49-4467-a2dd-68dc957ba1ed.mp4

- Ensure clicking menu items shows the associated content
- Ensure clicking outside the menu closes it

# Testing Bento mode
1. Run `amp dist`
2. Run `amp`
3. Navigate to `./test/fixtures/e2e/bento/mega-menu.html` (eg http://localhost:8000/test/fixtures/e2e/bento/mega-menu.html)

- Ensure menu behaves the same as the React mode
- Ensure the "light DOM" is manipulated
--  The `<section>` should have the `expanded` attribute set appropriately
-- The Title and Contents for each section should have `aria-*` attributes set appropriately
- Ensure clicking "outside" closes the menu

https://user-images.githubusercontent.com/430608/168639588-bda20903-f7d8-4f07-9a2c-94413e1967f3.mp4

![image](https://user-images.githubusercontent.com/430608/168642224-e45a80b2-448a-4699-8632-710aba4dc0f1.png)
